azero/contracts/migrations/lib.rs

```diff
@@ -0,0 +1,54 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+#[ink::contract]
+mod migrations {
+
+    use ink::{
+        env::Error as InkEnvError,
+        prelude::{format, string::String},
+    };
+    use scale::{Decode, Encode};
+
+    #[derive(Debug, PartialEq, Eq, Encode, Decode)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum MigrationsError {
+        InkEnvError(String),
+    }
+
+    impl From<InkEnvError> for MigrationsError {
+        fn from(why: InkEnvError) -> Self {
+            Self::InkEnvError(format!("{:?}", why))
+        }
+    }
+
+    #[ink(storage)]
+    pub struct Migrations {
+        last_completed_migration: u32,
+        owner: AccountId,
+    }
+
+    impl Migrations {
+        #[ink(constructor)]
+        pub fn new(owner: AccountId) -> Self {
+            Self {
+                last_completed_migration: 0,
+                owner,
+            }
+        }
+
+        #[ink(message)]
+        pub fn set_completed(&mut self, completed: u32) {
```

Comment:
I think there is an "is_owner" check missing

Reply:
oh yeah, very good catch!